src/SfxWeb/src/app/views/cluster/federation/federation.component.ts

```diff
@@ -0,0 +1,656 @@
+      nodeSpacing: 100
+    }
+  },
+  nodes: {
+    shape: "circle",
+    font: {
+      size: 32,
+      color: "#000000",
+    },
+    color: {
+      border: "rgba(10,141,16,1)",
+      background: "rgba(124,255,131,1)",
+      highlight: {
+        border: "rgba(10,141,16,1)",
+        background: "rgba(0,255,1,1)",
+      },
+      hover: {
+        border: "rgba(10,141,16,1)",
+        background: "rgba(0,255,1,1)",
+      }
+    },
+    borderWidth: 2,
+  },
+  edges: {
+    width: 1,
+    hoverWidth: 10,
+    color: {
+      color: "rgba(10,141,16,0.3)",
+      hover: "rgba(10,141,16,1)"
+    }
+  },
+  interaction: {
+    tooltipDelay: 0,
+    hideEdgesOnDrag: true,
+    hideEdgesOnZoom: true,
+    hover: true
+  }
+};
+
+@Component({
+  selector: 'app-federation',
+  templateUrl: './federation.component.html',
+  styleUrls: ['./federation.component.scss']
+})
+export class FederationComponent extends BaseControllerDirective implements OnInit, OnDestroy {
+  // nodeEvents
+  eventStoreHandler: IEventStoreData<NodeEventList, NodeEvent>;
+  eventStoreHandlerSubscription: Subscription;
+  // partitionEvents, but only the FM related
+  eventStoreHandlerFM: IEventStoreData<PartitionEventList, PartitionEvent>;
+  eventStoreHandlerFMSubscription: Subscription;
+  failedToLoadEvents: boolean;
+
+  // timeline bar and dates related settings
+  optionsConfig: IOptionConfig;
+  private debounceHandler: Subject<IOnDateChange> = new Subject<IOnDateChange>();
+  private debouncerHandlerSubscription: Subscription;
+  public startDateMin: Date;
+  public startDateMax: Date;
+  public startDate: Date;
+  public endDate: Date;
+  public quickDates = [
+    { display: 'Last 1 Hour', hours: 1 },
+    { display: 'Last 3 Hours', hours: 3 },
+    { display: 'Last 6 Hours', hours: 6 },
+    { display: 'Last 1 Day', hours: 24 },
+    { display: 'Last 7 Days', hours: 168 }
+  ];
+
+  constructor(protected data: DataService, injector: Injector, private telemService: TelemetryService) {
+    super(injector);
+  }
+
+  ngOnInit() {
+    super.ngOnInit();
+
+    // timeline bar config
+    this.optionsConfig = {
+      enableNodes: true,
+      enableRepairTasks: true
+    };
+
+    // setting up the timeline bar
+    // debounceTime is wanted delay for frequent bar changes
+    this.resetSelectionProperties();
+    this.debouncerHandlerSubscription = this.debounceHandler
+      .pipe(debounceTime(400), distinctUntilChanged())
+      .subscribe(dates => {
+        this.startDate = new Date(dates.startDate);
+        this.endDate = new Date(dates.endDate);
+    });
+  }
+
+  ngOnDestroy() {
+    // free the resources
+    this.debouncerHandlerSubscription.unsubscribe();
+    // this.eventStoreHandlerFMSubscription.unsubscribe();
+    // this.eventStoreHandlerSubscription.unsubscribe();
+  }
+
+  // represent selected node's neighborhood
+  onChangeInput(event: any){
+    let hie: HTMLInputElement = document.getElementById("byName") as HTMLInputElement;
```

Comment:
Same for this and all the following, use data binding. See: https://angular.io/guide/two-way-binding, and 
 You can also find some existing examples in this repo, src/app/modules/powershell-commands may have some that could be helpful.

Reply:
I'm not sure if you meant exactly the two way binding, but if you did it didn't turn out very well when I tried to implement it (maybe because I'm doing it bad), especially when I used it to handle radio button logics. in this case, [(ngModel)] has to have special [value] for every button in here which makes whole implementation more difficult and unreadable, in comparison where in current code I just have to handle 3 lines in order for it to work. also, I had big problems with dispatching events since it looks like two way binding has kind of delay and in SFX it looks like the code is bugged. in the repo you've provided I couldn't actually see where this two way is being used, since it is only declared in template view but doesn't appear in .ts file

Reply:
I would say to look into the angular forms for this https://www.positronx.io/angular-radio-buttons-example/ This would also let you have a set up where you know when by subscribing to the value changes event.